packages/grid/x-grid/src/tests/events.XGrid.test.tsx

```diff
@@ -287,4 +298,21 @@ describe('<XGrid /> - Events Params', () => {
     gridWindow.dispatchEvent(new Event('scroll'));
     expect(handleOnRowsScrollEnd.callCount).to.equal(1);
   });
+
+  it('call onViewportRowsChange when the viewport rows change', () => {
+    const handleOnViewportRowsChange = spy();
+    const { container } = render(
+      <div style={{ width: 300, height: 300 }}>
+        <TestVirtualization onViewportRowsChange={handleOnViewportRowsChange} />
+      </div>,
+    );
+    const gridWindow = container.querySelector('.MuiDataGrid-window');
+    // scroll 10 rows.
+    gridWindow.scrollTop = 12345;
```

Comment:
We could call `scrollToIndexes` instead.

Reply:
The risk would be that `scrollToIndexes` doesn't behave correctly, or we don't understand how it behaves at a deep enough level to write/read the test cae.

Reply:
I updated that based on the changed functionality